tests/cpp/solvers_interface/test_exceptions.cpp

```diff
@@ -2,157 +2,211 @@
+                     matvalRow.data(),
+                     {});
+
+    return solver;
 }
 
-TEST_CASE("InvalidColTypeException", "[exceptions][invalid_col_type]") {
-  for (auto const& solver_name : {"CBC", "CLP"}) {
-    SolverAbstract::Ptr solver = createSimpleProblem(solver_name);
-    try {
-      solver->chg_col_type(std::vector<int>(1, 0), std::vector<char>(1, 'P'));
-    } catch (InvalidColTypeException& ex) {
-      REQUIRE(ex.ErrorMessage() == "Invalid col type P for solver.");
+TEST_CASE("InvalidColTypeException", "[exceptions][invalid_col_type]")
+{
+    for (const auto& solver_name: {"CBC", "CLP"})
+    {
+        std::shared_ptr<SolverAbstract> solver = createSimpleProblem(solver_name);
+        try
+        {
+            solver->chg_col_type(std::vector<int>(1, 0), std::vector<char>(1, 'P'));
+        }
+        catch (InvalidColTypeException& ex)
+        {
+            REQUIRE(ex.ErrorMessage() == "Invalid col type P for solver.");
+        }
     }
-  }
 }
 
-TEST_CASE("InvalidBoundTypeException", "[exceptions][invalid_bound_type]") {
-  for (auto const& solver_name : {"CBC", "CLP"}) {
-    SolverAbstract::Ptr solver = createSimpleProblem(solver_name);
-
-    try {
-      solver->chg_bounds(std::vector<int>(1, 0), std::vector<char>(1, 'P'),
-                         std::vector<double>(1, 0));
-    } catch (InvalidBoundTypeException& ex) {
-      REQUIRE(ex.ErrorMessage() == "Invalid bound type P for solver.");
+TEST_CASE("InvalidBoundTypeException", "[exceptions][invalid_bound_type]")
+{
+    for (const auto& solver_name: {"CBC", "CLP"})
+    {
+        std::shared_ptr<SolverAbstract> solver = createSimpleProblem(solver_name);
+
+        try
+        {
+            solver->chg_bounds(std::vector<int>(1, 0),
+                               std::vector<char>(1, 'P'),
+                               std::vector<double>(1, 0));
+        }
+        catch (InvalidBoundTypeException& ex)
+        {
+            REQUIRE(ex.ErrorMessage() == "Invalid bound type P for solver.");
+        }
     }
-  }
 }
 
-TEST_CASE("InvalidSetAlgorithm", "[exceptions][invalid_set_algorithm]") {
-  for (auto const& solver_name : {"CBC", "CLP"}) {
-    SolverFactory factory;
-    SolverAbstract::Ptr solver = factory.create_solver(solver_name);
-
-    try {
-      solver->set_algorithm("BARRIER");
-    } catch (InvalidSolverOptionException& ex) {
-      REQUIRE(ex.ErrorMessage() ==
-              "Invalid option 'set_algorithm : BARRIER' for solver.");
+TEST_CASE("InvalidSetAlgorithm", "[exceptions][invalid_set_algorithm]")
+{
+    for (const auto& solver_name: {"CBC", "CLP"})
+    {
+        SolverFactory factory;
+        std::shared_ptr<SolverAbstract> solver = factory.create_solver(solver_name);
+
+        try
+        {
+            solver->set_algorithm("BARRIER");
+        }
+        catch (InvalidSolverOptionException& ex)
+        {
+            REQUIRE(ex.ErrorMessage() == "Invalid option 'set_algorithm : BARRIER' for solver.");
+        }
```

Comment:
Note this code doesn't actually check that an exception is thrown. It only checks that if an exception is thrown, the error message is as expected.